### PR TITLE
Fix for FlexBridge LT-16597, instructions blank on Linux

### PR DESCRIPTION
**Pull request by jasonleenaylor**
* Navigate control to temp html file after the form is shown
* Disable browser shortcuts and right click menu items

### CODE REVIEW

File: src/Chorus/UI/Settings/SendReceiveSettings.resx [hunk 1]
```diff
@@ -1,141 +1,142 @@
...
+  </data>
+  <data name="_helpProvider.TrayLocation" type="System.Drawing.Point, System.Drawing">
+    <value>17, 17</value>
+  </data>
+  <data name="ChorusHubSetupInstructionsHTML" xml:space="preserve">
+    <value>&lt;!DOCTYPE html&gt;&lt;html&gt;&lt;body style='font-family:"Microsoft Sans Serif","sans-serif";font-size:9pt'&gt;
 &lt;p style='margin:0in;'&gt;Chorus Hub is a simple way to Send/Receive on a local network, like in an office or home. These instructions are basic, but they should be enough to get you started.&lt;/p&gt;
 &lt;ol&gt;
-&lt;li style='margin-left:-10pt'&gt;Designate one computer on your network as &quot;the server&quot;, and use the following instructions to make sure this product is installed and running on it.&lt;/li&gt;
+&lt;li style='margin-left:-10pt'&gt;Designate one computer on your network as "the server", and use the following instructions to make sure this product is installed and running on it.&lt;/li&gt;
```
Comment (marksvc) [inline src/Chorus/UI/Settings/SendReceiveSettings.resx]: These quotes don't cause a problem?

File: src/Chorus/UI/Settings/SendReceiveSettings.resx [hunk 2]
```diff
@@ -1,141 +1,142 @@
...
 &lt;li style='list-style-type:lower-alpha;margin-left:-10pt'&gt;On Linux, Chorus Hub is automatically installed when you install FLEx. Use the Dash or Main Menu to find and launch Chorus Hub.&lt;/li&gt;
 &lt;/ol&gt;&lt;/li&gt;
-&lt;li style='margin-left:-10pt'&gt;Above these instructions, there is a checkbox labeled &quot;Show Chorus Hub as a Send/Receive option&quot;. Make sure this checkbox is selected.&lt;/li&gt;
+&lt;li style='margin-left:-10pt'&gt;Above these instructions, there is a checkbox labeled "Show Chorus Hub as a Send/Receive option". Make sure this checkbox is selected.&lt;/li&gt;
```
Comment (marksvc) [inline src/Chorus/UI/Settings/SendReceiveSettings.resx]: And here.

File: src/Chorus/UI/Settings/SendReceiveSettings.resx [hunk 3]
```diff
@@ -1,141 +1,142 @@
...
 &lt;/ol&gt;
-&lt;p style='margin:0in;'&gt;If all goes well, there is nothing more to do! Now from any other computer on the network, when you do Send/Receive, there should be a button labeled &quot;Chorus Hub&quot;, and under it should be a message telling you it has found Chorus Hub on the machine it is running on. If you click that button, your project will be Sent/Received.&lt;/p&gt;
+&lt;p style='margin:0in;'&gt;If all goes well, there is nothing more to do! Now from any other computer on the network, when you do Send/Receive, there should be a button labeled "Chorus Hub", and under it should be a message telling you it has found Chorus Hub on the machine it is running on. If you click that button, your project will be Sent/Received.&lt;/p&gt;
```
Comment (marksvc) [inline src/Chorus/UI/Settings/SendReceiveSettings.resx]: (And here)

File: src/Chorus/UI/Settings/SendReceiveSettings.cs [hunk 4]
```diff
@@ -20,10 +24,25 @@ public SendReceiveSettings()
...
 		}
 
+		/// <summary>
+		/// Make sure that the html content is displayed properly on both linux and windows by using Navigate on the chorusHubSetup control after the dialog is shown
+		/// </summary>
+		private void SendReceiveSettingsShown(object sender, EventArgs e)
+		{
+			System.ComponentModel.ComponentResourceManager resources = new System.ComponentModel.ComponentResourceManager(typeof(SendReceiveSettings));
+			var hubSetupText = LocalizationManager.GetString(@"Chorus_ChorusHubSetupInstructionsHtml", resources.GetString(@"ChorusHubSetupInstructionsHTML"),
+				@"Instructions shown before first Send/Receive. Please keep HTML tags as-is to preserve formatting, and use HTML escapes");
```
Comment (marksvc) [inline src/Chorus/UI/Settings/SendReceiveSettings.cs]: (I didn't entirely understand why the "Please keep HTML tags..." line was moved into here from the resource file, but okay.)

**Comment by marksvc** [comment]
I don't see where right-click menu items would be being turned off. Did you forget about that or did I just not find it in here?

Other than that and the embedded comments, it looks reasonable.

File: src/Chorus/UI/Settings/SendReceiveSettings.cs (hunk 4 above)
Comment (papeh) [inline src/Chorus/UI/Settings/SendReceiveSettings.cs]: Chorus handles localization differently than FieldWorks, so we write our notes to localizers in a different place. I put them in the wrong place to begin with, so Jason moved them to the right place.

File: src/Chorus/UI/Settings/SendReceiveSettings.cs [hunk 5]
```diff
@@ -76,7 +95,7 @@ private void internetCheckChanged(object sender, EventArgs e)
 		}
 
 		private void _helpButton_Click(object sender, EventArgs e)
-		{
+		{
```
Comment (papeh) [inline src/Chorus/UI/Settings/SendReceiveSettings.cs]: trailing whitespace

File: src/Chorus/UI/Settings/SendReceiveSettings.cs [hunk 6]
```diff
@@ -20,10 +24,25 @@ public SendReceiveSettings()
...
 
+		/// <summary>
+		/// Make sure that the html content is displayed properly on both linux and windows by using Navigate on the chorusHubSetup control after the dialog is shown
+		/// </summary>
+		private void SendReceiveSettingsShown(object sender, EventArgs e)
+		{
+			System.ComponentModel.ComponentResourceManager resources = new System.ComponentModel.ComponentResourceManager(typeof(SendReceiveSettings));
+			var hubSetupText = LocalizationManager.GetString(@"Chorus_ChorusHubSetupInstructionsHtml", resources.GetString(@"ChorusHubSetupInstructionsHTML"),
+				@"Instructions shown before first Send/Receive. Please keep HTML tags as-is to preserve formatting, and use HTML escapes");
+			var tempFile = Path.Combine(Path.GetTempPath(), Path.ChangeExtension(Path.GetRandomFileName(), ".html"));
```
Comment (papeh) [inline src/Chorus/UI/Settings/SendReceiveSettings.cs]: using?

File: src/Chorus/UI/Settings/SendReceiveSettings.cs (hunk 4 above)
Comment (jasonleenaylor) [inline src/Chorus/UI/Settings/SendReceiveSettings.cs]: The localization of the content will not be done by people who look at the resource file. So the comment in the resource file informs people to look elsewhere to localize, and the instructions for HTML are moved into a comment that shows in the tool which will be used.

File: src/Chorus/UI/Settings/SendReceiveSettings.resx (hunk 1 above)
Comment (jasonleenaylor) [inline src/Chorus/UI/Settings/SendReceiveSettings.resx]: Well, VisualStudio did the unescaping without my input, and it appears that the content displays correctly. Quotes in html content don't need to be escaped.

**Comment by marksvc** [comment]
So far LGTM other than right-click.

File: src/Chorus/UI/Settings/SendReceiveSettings.Designer.cs [hunk 7]
```diff
@@ -192,14 +191,16 @@ private void InitializeComponent()
...
-			| System.Windows.Forms.AnchorStyles.Left)
-			| System.Windows.Forms.AnchorStyles.Right)));
+			// 
+			this.chorusHubSetup.AllowWebBrowserDrop = false;
+			this.chorusHubSetup.Anchor = ((System.Windows.Forms.AnchorStyles)((((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Bottom) 
+            | System.Windows.Forms.AnchorStyles.Left) 
+            | System.Windows.Forms.AnchorStyles.Right)));
 			this.chorusHubSetup.BackColor = System.Drawing.Color.White;
 			this.chorusHubSetup.Font = new System.Drawing.Font("Segoe UI", 9F);
+			this.chorusHubSetup.IsWebBrowserContextMenuEnabled = false;
```
Comment (jasonleenaylor) [inline src/Chorus/UI/Settings/SendReceiveSettings.Designer.cs]: This is where the right click menu is disabled.

**Pull request merged**